effekt/jvm/src/main/scala/effekt/Server.scala

```diff
@@ -397,6 +397,32 @@ class Server(config: EffektConfig, compileOnChange: Boolean=false) extends Langu
             inlayHint.setData("capture")
             inlayHint
         }.toVector
+
+        val unannotated = getTreesInRange(range)(using context).map { trees =>
+          trees.collect {
+            // Functions without an annotated type:
+            case fun: FunDef if fun.ret.isEmpty => for {
+              sym <- context.symbolOption(fun.id)
+              tpe <- context.functionTypeOption(sym)
+              pos = fun.ret.span.range.from
+            } yield {
+              val prettyType = pp": ${tpe.result} / ${tpe.effects}"
+              val inlayHint = new InlayHint(convertPosition(pos), messages.Either.forLeft(pp": ${tpe.result} / ${tpe.effects}"))
+              inlayHint.setKind(InlayHintKind.Type)
+              val markup = new MarkupContent()
+              markup.setValue(s"return type${prettyType}")
+              markup.setKind("markdown")
+              inlayHint.setTooltip(markup)
+              inlayHint.setPaddingLeft(true)
+              inlayHint.setData("omitted-type-annotation")
```

Comment:
Philosophically, I think it would be wise to name this as to what it adds, not what it resolves. :)
(I'm willing to compromise about this, but let's argue first if you feel strongly about `omitted-`)
```suggestion
              inlayHint.setData("return-type-annotation")
```
(don't forget the tests too)